Content.Shared/Bed/SharedBedSystem.cs

```diff
@@ -56,8 +56,13 @@ private void OnStrapped(Entity<HealOnBuckleComponent> bed, ref StrappedEvent arg
 
     private void OnUnstrapped(Entity<HealOnBuckleComponent> bed, ref UnstrappedEvent args)
     {
-        _actionsSystem.RemoveAction(args.Buckle.Owner, bed.Comp.SleepAction);
-        _sleepingSystem.TryWaking(args.Buckle.Owner);
+        // If the entity being unbuckled is terminating, we shouldn't try to act upon it, as some components may be gone
+        if (!Terminating(args.Buckle.Owner))
+        {
+            _actionsSystem.RemoveAction(args.Buckle.Owner, bed.Comp.SleepAction);
+            _sleepingSystem.TryWaking(args.Buckle.Owner);
+        }
```

Comment:
```suggestion
        if (Terminating(args.Buckle.Owner))
            return;

        _actionsSystem.RemoveAction(args.Buckle.Owner, bed.Comp.SleepAction);
        _sleepingSystem.TryWaking(args.Buckle.Owner);
```
just do it like this